test/Conversion/LLHDToLLVM/signal-init.mlir

```diff
@@ -10,13 +10,11 @@ llhd.entity @root() -> () {
 llhd.entity @initSig () -> () {
   %1 = hw.aggregate_constant [ 0 : i1, 1 : i1] : !hw.array<2xi1>
   %2 = hw.aggregate_constant [ 0 : i1, 1 : i5] : !hw.struct<f1: i1, f2: i5>
-  // CHECK: [[V0:%.+]] = llvm.mlir.addressof @{{.+}} : !llvm.ptr<array<2 x i1>>
-  // CHECK: [[V1:%.+]] = llvm.load [[V0]] : !llvm.ptr<array<2 x i1>>
-  // CHECK: llvm.store [[V1]], {{%.+}} : !llvm.ptr<array<2 x i1>>
```

Comment:
It seems a bit strange to me that this store operation (and the two further down) are gone now, even though the code that inserts those ops was not touched. 🤔
@fabianschuiki Do you have a clue?

For reference: https://github.com/llvm/circt/blob/2f8381cd1bdf4411e5354038258a17bd717b7130/lib/Conversion/LLHDToLLVM/LLHDToLLVM.cpp#L1182C78-L1182C78

Reply:
Hmm yeah. Maybe some interaction with canonicalization or other conversion patterns in the lowering? Since it doesn't seem to break any of the simulations themselves, I'd say let's land this and then add an issue to check if this is an actual issue or just some canonicalization or similar suddenly cleaning things up.